adb_graphics/figures/maps.py

```diff
@@ -301,11 +302,31 @@ def draw(self, show=False):
         # correspond to a full field of contours.
         if self.hatch_fields:
             for field in self.hatch_fields:
-                self._draw_field(ax=ax,
-                                 field=field,
-                                 func=self.map.m.contourf,
-                                 **field.contour_kwargs,
-                                 )
+                colors = field.contour_kwargs.pop('colors')
+
+                cf = self._draw_field(ax=ax,
+                                      field=field,
+                                      func=self.map.m.contourf,
+                                      **field.contour_kwargs,
+                                      )
+
+                # For each level, we set the color of its hatch
+                for i, collection in enumerate(cf.collections):
+                    collection.set_edgecolor(colors[i % len(colors)])
+                    collection.set_facecolor(['None'])
+                    collection.set_linewidth(0.1)
+
+            # Create legend for precip type field
+            if self.field.short_name == 'ptyp':
+                plt.legend(handles=[mpatches.Patch(edgecolor='blue', facecolor='lightgrey', \
+                                                   label='Snow', hatch='----'), \
+                                    mpatches.Patch(edgecolor='red', facecolor='lightgrey', \
+                                                   label='Freezing Rain', hatch='\\\\\\\\'), \
+                                    mpatches.Patch(edgecolor='green', facecolor='lightgrey', \
+                                                   label='Rain', hatch='|||'), \
+                                    mpatches.Patch(edgecolor='purple', facecolor='lightgrey', \
+                                                   label='Ice Pellets', hatch='////')], \
+                                    loc=[0.25, 0.03])
```

Comment:
There's quite a bit of duplicate information between here and the default_specs.yml file. 

You could handle the patches in a list as you loop over hatch_fields like this:
```
colors = field.contour_kwargs.pop('colors')
label = field.contour_kwargs.pop('label')
...
handles.append(mpatches.Patch(edgecolor=colors, facecolor='lightgrey', label=label, 
                             hatch=field.contour_kwargs.get('hatches')[-1])
...
if self.field.short_name == 'ptyp':
    plt.legend(handles=handles, loc=[0.25, 0.03])


```

You'd need to make an empty list for handles outside that loop: 

```
handles = []
```

Reply:
Thanks for your comments. Quick responses here, we can discuss in depth at the weekly mtg.

- the linewidths are all leftovers from testing, I had trouble figuring out widths of contour outlines vs widths of hatches.  I'll clean those up.

-  I'll add the {grid} templates to the new fields in default_specs

- I had read while looking at this that ordered dictionaries were now the default (starting with v3.7?), but can add the dashes to make sure, and match the hatch and contour orders.

- I was able to get the hatches color to one item, the hatches to two, but couldn't get the levels to work with less than three.  I can look more into why that is, maybe just a magic combination I haven't found yet.

- I had a loop over hatches/patches in my first commit (when we tested the new linter with intentional errors). To prevent the "TooManyLocalVariables" pylint error, I'll probably have to move this into its own method.